homeassistant/components/samsungtv/bridge.py

```diff
@@ -313,26 +331,25 @@ async def async_mac_from_device(self) -> str | None:
         return mac_from_device_info(info) if info else None
 
     async def async_get_app_list(self) -> dict[str, str] | None:
-        """Get installed app list."""
-        return await self.hass.async_add_executor_job(self._get_app_list)
-
-    def _get_app_list(self) -> dict[str, str] | None:
         """Get installed app list."""
         if self._app_list is None:
-            if remote := self._get_remote():
-                raw_app_list: list[dict[str, str]] = remote.app_list()
-                self._app_list = {
-                    app["name"]: app["appId"]
-                    for app in sorted(raw_app_list, key=lambda app: app["name"])
-                }
+            if remote := await self._async_get_remote():
+                await remote.send_command(ChannelEmitCommand.get_installed_app())
+                response = process_api_response(await remote.connection.recv())
+                if response.get("data"):
+                    data = response["data"]
```

Comment:
Ideally more of this block moves into the library so it handles the timeout

Reply:
```python3

    async def async_get_app_list(self) -> dict[str, str] | None:
        """Get installed app list."""
        if self._app_list is not None:
            return self._app_list

        if not (remote := await self._async_get_remote()):
            return None

        async with async_timeout.timeout(TIMEOUT_WEBSOCKET):
            await remote.send_command(ChannelEmitCommand.get_installed_app())
            response = process_api_response(await remote.connection.recv())
            if not response.get("data"):
                return None

            data = response["data"]
            if not isinstance(data, dict) or not data.get("data"):
                return None

            raw_app_list = data["data"]
            self._app_list = {
                app["name"]: app["appId"]
                for app in sorted(raw_app_list, key=lambda app: cast(str, app["name"]))
            }

        return self._app_list
```

Adding a timeout here allowed it to startup

Reply:
This has caused a drop in coverage... but I think we can re-increase coverage later.